### ISSUE TITLE
Confusion about compute_dist_loss? Didn't consider rank?

### ISSUE THREAD

**Issue by beckhamchen** [facebookresearch/OTTER#3]
def compute_dist_loss(self, img_logits_s, txt_logits_s, sims):
        """
        Computes distillation loss computed over image and text logits.
        """
        # Compute the cost matrix. Add large values to the
        # diagonal of the cost matrix to ensure the output of sinkhorn is
        # close to 0 on the diagonal. The is because OTTER uses sinkhorn
        # to model off-diagonal target probabilities.
        diag = (torch.eye(*sims["vv_sim"].shape) * self.remove_diag * 1e2).to(img_logits_s)
        vv_sim = (sims["vv_sim"] - diag) * self.vv_coef
        tt_sim = (sims["tt_sim"] - diag) * self.tt_coef
        vt_sim = sims["vt_sim"]
        tv_sim = sims["tv_sim"]
        img_cost_mat = - (vv_sim + tt_sim + vt_sim)
        txt_cost_mat = - (vv_sim + tt_sim + tv_sim)

 Is here something wrong? diag = (torch.eye(*sims["vv_sim"].shape) * self.remove_diag * 1e2).to(img_logits_s)
Didn't consider rank when training using distributional-training? 
maybe 

            diag_square = torch.eye(sims["vv_sim"].shape[0])
            diag = torch.zeros(*sim["vv_sim"].shape)
            bs = sim["vv_sim"].shape[0]
            rank = get_rank()
            diag[:, rank * bs: (rank + 1) * bs] = diag_square
            diag = (diag * self.remove_diag * 1e2).to(img_logits_s)
